### PR TITLE
feat(section): #MAG-118 add section to cards controller

### PR DESCRIPTION
**Pull request by Sofianebernoussi**
## Describe your changes

added section to cards controller (remove side effect due to sections)

## Checklist tests

- Check if sections are correctly deleted when we delete a board
- Check if if cards are correctly moved/duplicated in a layout board
- Check if cards are correctly moved/deleted when we delete a section

## Issue ticket number and link
https://jira.support-ent.fr/browse/MAG-118

- [X] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java [hunk 1]
```diff
@@ -262,9 +262,13 @@ public Future<JsonObject> preDeleteBoards(String userId, List<String> boardIds,
...
         this.deleteBoards(userId, boardIds)
-                .compose(success -> this.folderService.updateOldFolder(boardIds))
+                .compose(success -> {
+                    Future<JsonObject> deleteSectionsFuture = this.sectionService.deleteByBoards(boardIds);
+                    Future<JsonObject> updateFolderFuture = this.folderService.updateOldFolder(boardIds);
+                    return CompositeFuture.all(deleteSectionsFuture, updateFolderFuture);
+                })
                 .onFailure(promise::fail)
-                .onSuccess(promise::complete);
+                .onSuccess(success -> promise.complete(new JsonObject()));
```
Comment (aymericdevriesere) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java:271]: Un petit message de succès ?

File: src/main/java/fr/cgi/magneto/controller/CardController.java [hunk 2]
```diff
@@ -208,47 +205,97 @@ public void duplicate(HttpServerRequest request) {
...
+                CompositeFuture.all(updateCardFuture, getBoardFuture, getOldBoardFuture, getSectionFuture, getOldSectionFuture)
                         .compose(result -> {
-                            Board currentBoard = getBoardFuture.result().get(0);
-                            BoardPayload boardToUpdate = new BoardPayload()
-                                    .setId(currentBoard.getId())
-                                    .setPublic(currentBoard.isPublic())
-                                    .setCardIds(currentBoard.cards()
+                            if (!getOldBoardFuture.result().isEmpty() && !getBoardFuture.result().isEmpty()) {
+                                List<Future> updateBoardsFutures = new ArrayList<>();
+                                Board currentBoard = getBoardFuture.result().get(0);
```
Comment (aymericdevriesere) [inline src/main/java/fr/cgi/magneto/controller/CardController.java:232]: T'es sur que tu récupères toujours quelque chose ?

File: src/main/java/fr/cgi/magneto/controller/CardController.java [hunk 3]
```diff
@@ -208,47 +205,97 @@ public void duplicate(HttpServerRequest request) {
...
+                                Board oldBoard = getOldBoardFuture.result().get(0);
+
+                                // Add cards in current board
+                                if (currentBoard.isLayoutFree()) {
```
Comment (aymericdevriesere) [inline src/main/java/fr/cgi/magneto/controller/CardController.java:236]: tu pourrais pas découper les if, else ? ça fait une grand fonction

File: src/main/java/fr/cgi/magneto/controller/CardController.java (hunk 2 above)
Comment (Sofianebernoussi) [reply src/main/java/fr/cgi/magneto/controller/CardController.java:232]: je verifie l230

**Review by flmartineau — APPROVED** [review]

**Pull request merged**